Xwt.Gtk/Xwt.GtkBackend/Gtk2IconEntry.cs

```diff
@@ -0,0 +1,264 @@
+					SetProperty ("primary-icon-tooltip-markup", val);
+				}
+			}
+		}
+
+		public Gdk.Pixbuf PrimaryIconPixbuf {
+			get {
+				using (GLib.Value val = GetProperty ("primary-icon-pixbuf")) {
+					return (Gdk.Pixbuf)val;
+				}
+			}
+			set {
+				using (GLib.Value val = new GLib.Value (value)) {
+					SetProperty ("primary-icon-pixbuf", val);
+				}
+			}
+		}
+
+		public string SecondaryIconName {
+			get {
+				using (GLib.Value val = GetProperty ("secondary-icon-name")) {
+					return (string)val;
+				}
+			}
+			set {
+				using (GLib.Value val = new GLib.Value (value)) {
+					SetProperty ("secondary-icon-name", val);
+				}
+			}
+		}
+
+		public bool SecondaryIconActivatable {
+			get {
+				using (GLib.Value val = GetProperty ("secondary-icon-activatable")) {
+					return (bool)val;
+				}
+			}
+			set {
+				using (GLib.Value val = new GLib.Value (value)) {
+					SetProperty ("secondary-icon-activatable", val);
+				}
+			}
+		}
+
+		public bool SecondaryIconSensitive {
+			get {
+				using (GLib.Value val = GetProperty ("secondary-icon-sensitive")) {
+					return (bool)val;
+				}
+			}
+			set {
+				using (GLib.Value val = new GLib.Value (value)) {
+					SetProperty ("secondary-icon-sensitive", val);
+				}
+			}
+		}
+
+		public string SecondaryIconTooltipText {
+			get {
+				using (GLib.Value val = GetProperty ("secondary-icon-tooltip-text")) {
+					return (string)val;
+				}
+			}
+			set {
+				using (GLib.Value val = new GLib.Value (value)) {
+					SetProperty ("secondary-icon-tooltip-text", val);
+				}
+			}
+		}
+
+		public string SecondaryIconTooltipMarkup {
+			get {
+				using (GLib.Value val = GetProperty ("secondary-icon-tooltip-markup")) {
+					return (string)val;
+				}
+			}
+			set {
+				using (GLib.Value val = new GLib.Value (value)) {
+					SetProperty ("secondary-icon-tooltip-markup", val);
+				}
+			}
+		}
+
+		public Gdk.Pixbuf SecondaryIconPixbuf {
+			get {
+				using (GLib.Value val = GetProperty ("secondary-icon-pixbuf")) {
+					return (Gdk.Pixbuf)val;
+				}
+			}
+			set {
+				using (GLib.Value val = new GLib.Value (value)) {
+					SetProperty ("secondary-icon-pixbuf", val);
+				}
+			}
+		}
+
+		[GLib.Signal("icon-press")]
+		public event Gtk.IconPressHandler IconPress {
+			add {
+				this.AddSignalHandler ("icon-press", value, typeof (Gtk.IconPressArgs));
+			}
+			remove {
+				this.RemoveSignalHandler ("icon-press", value);
+			}
+		}
+
+		[GLib.Signal ("icon-release")]
+		public event Gtk.IconReleaseHandler IconRelease {
+			add {
+				this.AddSignalHandler ("icon-release", value, typeof (Gtk.IconReleaseArgs));
+			}
+			remove {
+				this.RemoveSignalHandler ("icon-release", value);
+			}
+		}
+	}
+}
+
+namespace Gtk
+{
+	public delegate void IconPressHandler(object o, IconPressArgs args);
+
+	public delegate void IconReleaseHandler (object o, IconReleaseArgs args);
+
+	public class IconPressArgs : GLib.SignalArgs {
+		public Gtk.EntryIconPosition P0{
```

Comment:
It's from the Gtk generator, to make Gtk2 and 3 work the same way. The long mission is to add customizable icons for TextEntry.